### PR TITLE
Initial project structure

### PR DESCRIPTION
**Pull request by dgutride**
Includes:
- webpack for bundling, test and dev configurations
- sample component
- sample unit test
- gulp file for transpilation and build helpers
- starter demo site structure

### CODE REVIEW

File: config/webpack.demo.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
+const helpers = require('./helpers');
+const webpack = require('webpack');
+const path = require('path');
+const sass = require('./sass');
```
Comment (AdamJ) [inline config/webpack.demo.js]: Is this repo going to use the SASS version of PatternFly, or stick with the default LESS version from V3?

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      return file.base; // because of Angular 2's encapsulation, it's natural to save the css where the less file was
+    }));
+});
+
+//Less compilation and minifiction - adopted from sass compilation, needs work
+gulp.task('transpile-less', function () {
+  return transpileLESS(appSrc + '/app/**/*.less');
+});
+
+// Put the SASS files back to normal
```
Comment (AdamJ) [inline gulpfile.js]: Confused here (and related to above question) is this using both preprocessors?

Comment (dgutride) [reply gulpfile.js]: It is only going to be LESS - there is another issue to manage CSS (which this does not cover).  @mindreeper2420

File: config/webpack.demo.js (hunk 1 above)
Comment (dgutride) [reply config/webpack.demo.js]: This is an unused variable - see comment below and does not indicate any technology choice.

**Review by dlabrecq — APPROVED** [review]
LGTM

**Pull request merged**